hal/src/rtl872x/i2c_hal.cpp

```diff
@@ -188,15 +210,17 @@ class I2cClass {
         conf.mode = OUTPUT_OPEN_DRAIN_PULLUP;
         conf.set_value = true;
         conf.value = 1;
-        hal_gpio_configure(sdaPin_, &conf, nullptr);
-        conf.value = hal_gpio_read(sclPin_);
         hal_gpio_configure(sclPin_, &conf, nullptr);
+        hal_gpio_configure(sdaPin_, &conf, nullptr);
+
+        // Check if slave is stretching the SCL
+        if (!WAIT_TIMED(transConfig_.timeout_ms, hal_gpio_read(sclPin_) == 0)) {
```

Comment:
I'd use some default timeout here instead, perhaps based on the i2c clock speed e.g. x number of bits @ clock speed. The default 100ms is probably the best here.